internals/overlord/checkstate/manager.go

```diff
@@ -633,3 +644,50 @@ func (m *CheckManager) Replan() {
 		m.updateCheckData(check, changeID, 0)
 	}
 }
+
+// RefreshCheck runs a check immediately.
+func (m *CheckManager) RefreshCheck(ctx context.Context, check *plan.Check) (*CheckInfo, error) {
+	m.checksLock.Lock()
+	checkData := m.ensureCheck(check.Name)
+	refresh := checkData.refresh
+	result := checkData.result
+	m.checksLock.Unlock()
+
+	getCheckInfo := func() *CheckInfo {
+		m.checksLock.Lock()
+		checkData := m.ensureCheck(check.Name)
+		m.checksLock.Unlock()
+		info := CheckInfo{
+			Name:      checkData.name,
+			Level:     checkData.level,
+			Startup:   checkData.startup,
+			Status:    checkData.status,
+			Failures:  checkData.failures,
+			Threshold: checkData.threshold,
+			ChangeID:  checkData.changeID,
+		}
+		return &info
+	}
+
+	// If the check is stopped, run the check directly without using changes and tasks.
+	if checkData.changeID == "" {
+		chk := newChecker(check)
+		err := runCheck(ctx, chk, check.Timeout.Value)
+		return getCheckInfo(), err
+	}
+	if refresh == nil || result == nil {
+		panic(fmt.Sprintf("internal error: refresh channels not initialized for check %q", checkData.name))
+	}
+
+	select {
+	case refresh <- refreshInfo{ctx, result}:
+	case <-ctx.Done():
+		return getCheckInfo(), ctx.Err()
+	}
```

Comment:
Here's where we'd make the result channel:
```suggestion
	result := make(chan error)
	select {
	case refresh <- refreshInfo{ctx, result}:
	case <-ctx.Done():
		return getCheckInfo(), ctx.Err()
	}
```